### PR TITLE
Put javascript values into datastore

### PR DESCRIPTION
**Pull request by myyong**
It looks like the values are returned, without awaiting for the datastore to finish 'creating' and returning "Created." Do we anticipate a problem with this async?

This pull request doesn't yet contain work for eval-ing the rest of the code, only about putting the values onto datastore.

### CODE REVIEW

**Comment by myyong** [comment]
Actually I can see a problem there already. We need put the value as a Dataframe along with the URL.

**Comment by myyong** [comment]
@tpetricek

It seems to me that I need the PUT endpoint to datastore to return the URL of the value. This is so that I can send the correct URL to the imported frames in the python plugin.

Current: 
```
$ curl --data '[{"Col1":123, "Col2":"Abc"},{"Col1":456, "Col2":"Def"}]' 
    -X PUT http://localhost:7102/put/8efbb4d8f6f445f4851d39a7ffb281ef/one
Created
```

Need: 
```
$ curl --data '[{"Col1":123, "Col2":"Abc"},{"Col1":456, "Col2":"Def"}]' 
    -X PUT http://localhost:7102/put/8efbb4d8f6f445f4851d39a7ffb281ef/one
http://wrattler_wrattler_data_store_1:7102/8efbb4d8f6f445f4851d39a7ffb281ef/one
```

File: client/src/languagePlugins/javascript/javascriptPlugin.ts [hunk 1]
```diff
@@ -175,31 +175,28 @@ class JavascriptBlockKind implements Langs.Block {
...
+        let url = DATASTORE_URI.concat("/"+hash).concat("/"+variableName)
         let headers = {'Content-Type': 'application/json'}
-        let url = DATASTORE_URI.concat(pathname)
         try {
-          const response = await axios.get(url, {headers: headers});
-          console.log(response)
-          return response.data
+          var response = await axios.put(url, value, {headers: headers});
+          // return DATASTORE_URI.concat("/"+hash).concat("/"+variableName)
+          return "http://wrattler_wrattler_data_store_1:7102".concat("/"+hash).concat("/"+variableName)
```
Comment (tpetricek) [inline client/src/languagePlugins/javascript/javascriptPlugin.ts:185]: We need to figure out how to replace this with some sort of configuration. More comments below.

**Comment by tpetricek** [comment]
This looks great - the only missing piece is that I'm not sure how to handle do the URLs properly.

@nbarlowATI @myyong can the two of you meet to discuss this and figure this out?

Right now, our `docker-compose` file specifies (among other things) the following:

```
  wrattler_client:
    ports:
     - "8080:8080"
    environment:
     - PYTHONSERVICE_URI=http://localhost:7101
     - DATASTORE_URI=http://localhost:7102
     
  wrattler_python_service:
    ports:
    - "7101:7101"
    environment:
     - DATASTORE_URI=http://wrattler_wrattler_data_store_1:7102
```

The thing I do not understand is where do the server names such as `wrattler_wrattler_data_store_1` come from and when we should use `wrattler_wrattler_data_store_1` and when `localhost`. (And what will we need to do when we deploy this to Azure?)

I guess the difference is that python service and data store are both running inside the same Docker compose network, so they can talk to each other using the Docker-defined names. The client is hosted by Docker, but runs in my web browser, so it needs a URL that refers to the Docker system from the outside i.e. `localhost`.

When we pass URLs between client and Docker services, it looks like we get into trouble because we need different URLs (`localhost` for client and `wrattler_wrattler_data_store_1` inside Docker). 

Is there some way to use just one URL that would work both in client and inside Docker? If no, what should we do? (We could keep just relative URLs or do some sort of string replace to translate between internal and external.)

**Pull request merged**